kernel/src/lib/nuumshell/shell.cr

```diff
@@ -0,0 +1,66 @@
+require "./apps/*"
+
+class NuumShell
+  def initialize
+    @user = "root"
+    @pass = "root"
+    @login = false
+    @max_login = 3
+  end
+
+  def run
+    banner
+    while !@login
+      login
+    end
+    loop do
+      print "\n#{@user}@nuumsh#: "
+      command = Keyboard.gets.chomp
+      if command.starts_with?("echo")
+        Echo.echo(command)
+      end
+    end
+  end
+
+  def login
+    try = 0
+    loop do
+      print "Login: "
+      user = Keyboard.gets.chomp
+      print "Password: "
+      pass = Keyboard.gets(:silent).chomp
+      if pass == @pass && user == @user
+        @login = true
+        break
+      end
+      puts "User or password are incorrect"
+      try = +1
```

Comment:
This is most likely a bug. You are not incrementing `try`, you are only setting it to positive `1`.